barretenberg/cpp/src/barretenberg/dsl/acir_format/civc_recursion_constraints.test.cpp

```diff
@@ -0,0 +1,155 @@
+                               civc_data.proof.goblin_proof.eccvm_proof.pre_ipa_proof.begin(),
+                               civc_data.proof.goblin_proof.eccvm_proof.pre_ipa_proof.end());
+        proof_witnesses.insert(proof_witnesses.end(),
+                               civc_data.proof.goblin_proof.eccvm_proof.ipa_proof.begin(),
+                               civc_data.proof.goblin_proof.eccvm_proof.ipa_proof.end());
+        proof_witnesses.insert(proof_witnesses.end(),
+                               civc_data.proof.goblin_proof.translator_proof.begin(),
+                               civc_data.proof.goblin_proof.translator_proof.end());
+
+        // Construct witness indices for each component in the constraint; populate the witness array
+        auto [key_indices, key_hash_index, proof_indices, public_inputs_indices] =
+            ProofSurgeon<fr>::populate_recursion_witness_data(
+                program.witness,
+                proof_witnesses,
+                key_witnesses,
+                key_hash_witness,
+                /*num_public_inputs_to_extract=*/civc_data.mega_vk->num_public_inputs - PUBLIC_INPUTS_SIZE);
+
+        auto constraint = RecursionConstraint{ .key = key_indices,
+                                               .proof = proof_indices,
+                                               .public_inputs = public_inputs_indices,
+                                               .key_hash = key_hash_index,
+                                               .proof_type = PROOF_TYPE::CIVC };
+
+        // Construct a constraint system
+        program.constraints.varnum = static_cast<uint32_t>(program.witness.size());
+        program.constraints.num_acir_opcodes = static_cast<uint32_t>(1);
+        program.constraints.civc_recursion_constraints = { constraint };
+        program.constraints.original_opcode_indices = create_empty_original_opcode_indices();
+        mock_opcode_indices(program.constraints);
+
+        return program;
+    }
+
+    static std::shared_ptr<DeciderProvingKey> get_civc_recursive_verifier_pk(AcirProgram& program)
+    {
+        // Build constraints
+        Builder builder = create_circuit(program, { .honk_recursion = 2 });
+
+        info("Estimate finalized number of gates: ", builder.get_estimated_num_finalized_gates());
+
+        // Construct vk
+        auto proving_key = std::make_shared<DeciderProvingKey>(builder);
+
+        return proving_key;
+    }
+
+  protected:
+    static void SetUpTestSuite() { bb::srs::init_file_crs_factory(bb::srs::bb_crs_path()); }
+};
+
+TEST_F(CivcRecursionConstraintTest, GenerateRecursiveCivcVerifierVKFromConstraints)
+{
+    using VerificationKey = CivcRecursionConstraintTest::VerificationKey;
+    using ClientIVCData = CivcRecursionConstraintTest::ClientIVCData;
+
+    ClientIVCData civc_data = CivcRecursionConstraintTest::get_civc_data(TraceSettings());
+
+    std::shared_ptr<VerificationKey> actual_vk;
```

Comment:
Maybe this is a bit of a nit but I would expect the "actual" and "expected" labels to be reversed here. I think of the VK generated using real witnesses as the "ground truth" and the thing generated based on dummy witnesses as the thing we want to match that truth. I can see why you'd apply "actual" to the "ground truth" as well though so maybe these are just flawed terms. In any case, can you make it a bit more clear in the comments what exactly is being compared in these two tests?